osutil/user_test.go

```diff
@@ -36,6 +37,39 @@ type createUserSuite struct {
 
 var _ = check.Suite(&createUserSuite{})
 
+func (s *createUserSuite) TestAddExtraUserOnClassic(c *check.C) {
+	mockHome := c.MkDir()
+	restorer := osutil.MockUserLookup(func(string) (*user.User, error) {
+		current, err := user.Current()
+		if err != nil {
+			c.Fatalf("user.Current() failed with %s", err)
+		}
+		return &user.User{
+			HomeDir: mockHome,
+			Gid:     current.Gid,
+			Uid:     current.Uid,
+		}, nil
+	})
+	defer restorer()
+
+	mockAddUser := testutil.MockCommand(c, "adduser", "true")
+	defer mockAddUser.Restore()
+
+	restore := release.MockOnClassic(true)
```

Comment:
This way you won't have to mock release to test the user creation code. Less coupling is easier to work with.